// react-bootstrap-typeahead/cjs/components/MenuItem/BaseMenuItem.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Disabled = exports.Default = exports.Active = void 0;

var _react = _interopRequireDefault(require("react"));

var _MenuItem = require("./MenuItem");

/* eslint-disable sort-keys,import/no-extraneous-dependencies */
var _default = {
  title: 'Components/MenuItem/BaseMenuItem',
  component: _MenuItem.BaseMenuItem
};
exports["default"] = _default;

var Template = function Template(args) {
  return (
    /*#__PURE__*/
    // @ts-ignore
    _react["default"].createElement(_MenuItem.BaseMenuItem, args)
  );
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {
  children: 'This is a base menu item'
};
var Active = Template.bind({});
exports.Active = Active;
Active.args = {
  active: true,
  children: 'This is an active base menu item'
};
var Disabled = Template.bind({});
exports.Disabled = Disabled;
Disabled.args = {
  children: 'This is a disabled base menu item',
  disabled: true
};
