// react-bootstrap-typeahead/cjs/components/TypeaheadInputMulti/TypeaheadInputMulti.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.WithHint = exports.Valid = exports.Small = exports.Large = exports.Invalid = exports.FocusState = exports.Disabled = exports.Default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _Token = _interopRequireDefault(require("../Token"));

var _TypeaheadInputMulti = _interopRequireDefault(require("./TypeaheadInputMulti"));

var _data = _interopRequireDefault(require("../../tests/data"));

var _helpers = require("../../tests/helpers");

var _excluded = ["hintText"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var _default = {
  title: 'Components/TypeaheadInputMulti',
  component: _TypeaheadInputMulti["default"]
};
exports["default"] = _default;

var selected = _data["default"].slice(1, 4);

var defaultProps = {
  children: selected.map(function (option) {
    return /*#__PURE__*/_react["default"].createElement(_Token["default"], {
      key: option.name,
      option: option,
      onRemove: _helpers.noop
    }, option.name);
  }),
  selected: selected
};

var Template = function Template(_ref) {
  var _ref$hintText = _ref.hintText,
      hintText = _ref$hintText === void 0 ? '' : _ref$hintText,
      args = (0, _objectWithoutProperties2["default"])(_ref, _excluded);

  var _useState = (0, _react.useState)(args.value),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      value = _useState2[0],
      setValue = _useState2[1];

  var _useState3 = (0, _react.useState)(null),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      inputNode = _useState4[0],
      setInputNode = _useState4[1];

  return /*#__PURE__*/_react["default"].createElement(_helpers.HintProvider, {
    hintText: hintText,
    inputNode: inputNode
  }, /*#__PURE__*/_react["default"].createElement(_TypeaheadInputMulti["default"], (0, _extends2["default"])({}, args, {
    inputRef: setInputNode,
    onChange: function onChange(e) {
      return setValue(e.target.value);
    },
    referenceElementRef: _helpers.noop,
    value: value
  })));
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = _objectSpread({}, defaultProps);
var FocusState = Template.bind({});
exports.FocusState = FocusState;
FocusState.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  className: 'focus'
});
var Disabled = Template.bind({});
exports.Disabled = Disabled;
Disabled.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  children: selected.map(function (option) {
    return /*#__PURE__*/_react["default"].createElement(_Token["default"], {
      disabled: true,
      key: option.name,
      option: option,
      onRemove: _helpers.noop
    }, option.name);
  }),
  disabled: true
});
var Small = Template.bind({});
exports.Small = Small;
Small.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  size: 'sm'
});
var Large = Template.bind({});
exports.Large = Large;
Large.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  size: 'lg'
});
var Valid = Template.bind({});
exports.Valid = Valid;
Valid.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  className: 'focus',
  isValid: true
});
var Invalid = Template.bind({});
exports.Invalid = Invalid;
Invalid.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  className: 'focus',
  isInvalid: true
});
var WithHint = Template.bind({});
exports.WithHint = WithHint;
WithHint.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  hintText: 'california',
  value: 'cali'
});
