// react-bootstrap-typeahead/es/components/Menu/Menu.stories.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

/* eslint-disable sort-keys,import/no-extraneous-dependencies */
import React from 'react';
import Menu from './Menu';
import MenuItem from '../MenuItem';
import { noop } from '../../utils';
var options = [{
  label: 'Item 1'
}, {
  label: 'Item 2'
}, {
  label: 'Item 3'
}]; // TODO: Caused by `isRequiredForA11y` validator.
// @ts-ignore

export default {
  title: 'Components/Menu',
  component: Menu
};
var defaultProps = {
  inputHeight: 0,
  scheduleUpdate: noop,
  text: ''
};
var children = options.map(function (o, idx) {
  return /*#__PURE__*/React.createElement(MenuItem, {
    key: o.label,
    option: o,
    position: idx
  }, o.label);
});

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Menu, _extends({}, args, {
    style: _objectSpread(_objectSpread({}, args.style), {}, {
      position: 'relative'
    })
  }));
};

export var Default = Template.bind({});
Default.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  children: children,
  id: 'default-menu'
});
export var Empty = Template.bind({});
Empty.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  id: 'empty-menu'
});
export var HeaderAndDivider = Template.bind({});
HeaderAndDivider.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  children: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Menu.Header, null, "This is a menu header"), /*#__PURE__*/React.createElement(Menu.Divider, null), /*#__PURE__*/React.createElement(MenuItem, {
    option: options[0],
    position: 0
  }, options[0].label)),
  id: 'header-and-divider'
});
