// react-bootstrap-typeahead/cjs/components/Typeahead/Typeahead.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Pagination = exports.MultiSelect = exports.LoadingState = exports.InputGrouping = exports.Default = exports.CustomMenu = exports.Controlled = exports.ClearButton = exports.AllowNew = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _reactBootstrap = require("react-bootstrap");

var _Typeahead = _interopRequireDefault(require("./Typeahead"));

var _Menu = _interopRequireDefault(require("../Menu"));

var _MenuItem = _interopRequireDefault(require("../MenuItem"));

var _data = _interopRequireDefault(require("../../tests/data"));

var _helpers = require("../../tests/helpers");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var _default = {
  title: 'Components/Typeahead',
  component: _Typeahead["default"],
  argTypes: {
    align: {
      options: ['justify', 'left', 'right'],
      control: {
        type: 'radio'
      }
    },
    disabled: {
      control: {
        type: 'boolean'
      }
    },
    size: {
      options: ['default', 'sm', 'lg'],
      control: {
        type: 'radio'
      }
    }
  },
  parameters: {
    actions: {
      argTypesRegex: '^on.*'
    }
  }
};
exports["default"] = _default;
var defaultProps = {
  allowNew: false,
  clearButton: false,
  flip: true,
  id: 'rbt-id',
  isLoading: false,
  labelKey: 'name',
  multiple: false,
  onBlur: _helpers.noop,
  onChange: _helpers.noop,
  onFocus: _helpers.noop,
  onInputChange: _helpers.noop,
  onKeyDown: _helpers.noop,
  options: _data["default"],
  placeholder: 'Choose a state...',
  positionFixed: true
};

var Template = function Template(args) {
  return /*#__PURE__*/_react["default"].createElement(_Typeahead["default"], args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = _objectSpread({}, defaultProps);
var MultiSelect = Template.bind({});
exports.MultiSelect = MultiSelect;
MultiSelect.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  defaultSelected: _data["default"].slice(0, 4),
  multiple: true
});
var ClearButton = Template.bind({});
exports.ClearButton = ClearButton;
ClearButton.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  clearButton: true,
  defaultSelected: _data["default"].slice(0, 1)
});
var LoadingState = Template.bind({});
exports.LoadingState = LoadingState;
LoadingState.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  isLoading: true
});
var Pagination = Template.bind({});
exports.Pagination = Pagination;
Pagination.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  maxResults: 10,
  paginate: true
});
var AllowNew = Template.bind({});
exports.AllowNew = AllowNew;
AllowNew.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  allowNew: true
});
var CustomMenu = Template.bind({});
exports.CustomMenu = CustomMenu;
CustomMenu.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  renderMenu: function renderMenu(results, menuProps) {
    return /*#__PURE__*/_react["default"].createElement(_Menu["default"], (0, _extends2["default"])({}, menuProps, {
      inputHeight: 0,
      scheduleUpdate: _helpers.noop
    }), results.slice().reverse().map(function (r, index) {
      return (
        /*#__PURE__*/
        // @ts-ignore
        _react["default"].createElement(_MenuItem["default"], {
          key: r.name,
          option: r,
          position: index
        }, // @ts-ignore
        r.name)
      );
    }));
  }
});

var InputGrouping = function InputGrouping(args) {
  return /*#__PURE__*/_react["default"].createElement(_reactBootstrap.InputGroup, {
    size: args.size
  }, /*#__PURE__*/_react["default"].createElement(_reactBootstrap.InputGroup.Text, null, "$"), /*#__PURE__*/_react["default"].createElement(_Typeahead["default"], args), /*#__PURE__*/_react["default"].createElement(_reactBootstrap.InputGroup.Text, null, ".00"));
};

exports.InputGrouping = InputGrouping;
InputGrouping.args = _objectSpread({}, defaultProps);

var Controlled = function Controlled(args) {
  var _useState = (0, _react.useState)(args.selected || []),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      selected = _useState2[0],
      setSelected = _useState2[1];

  return /*#__PURE__*/_react["default"].createElement(_Typeahead["default"], (0, _extends2["default"])({}, args, {
    onChange: setSelected,
    selected: selected
  }));
};

exports.Controlled = Controlled;
Controlled.args = _objectSpread({}, defaultProps);
