// react-bootstrap-typeahead/cjs/components/Menu/Menu.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.HeaderAndDivider = exports.Empty = exports.Default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _Menu = _interopRequireDefault(require("./Menu"));

var _MenuItem = _interopRequireDefault(require("../MenuItem"));

var _utils = require("../../utils");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var options = [{
  label: 'Item 1'
}, {
  label: 'Item 2'
}, {
  label: 'Item 3'
}]; // TODO: Caused by `isRequiredForA11y` validator.
// @ts-ignore

var _default = {
  title: 'Components/Menu',
  component: _Menu["default"]
};
exports["default"] = _default;
var defaultProps = {
  inputHeight: 0,
  scheduleUpdate: _utils.noop,
  text: ''
};
var children = options.map(function (o, idx) {
  return /*#__PURE__*/_react["default"].createElement(_MenuItem["default"], {
    key: o.label,
    option: o,
    position: idx
  }, o.label);
});

var Template = function Template(args) {
  return /*#__PURE__*/_react["default"].createElement(_Menu["default"], (0, _extends2["default"])({}, args, {
    style: _objectSpread(_objectSpread({}, args.style), {}, {
      position: 'relative'
    })
  }));
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  children: children,
  id: 'default-menu'
});
var Empty = Template.bind({});
exports.Empty = Empty;
Empty.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  id: 'empty-menu'
});
var HeaderAndDivider = Template.bind({});
exports.HeaderAndDivider = HeaderAndDivider;
HeaderAndDivider.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  children: /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Menu["default"].Header, null, "This is a menu header"), /*#__PURE__*/_react["default"].createElement(_Menu["default"].Divider, null), /*#__PURE__*/_react["default"].createElement(_MenuItem["default"], {
    option: options[0],
    position: 0
  }, options[0].label)),
  id: 'header-and-divider'
});
