// react-bootstrap-typeahead/cjs/components/TypeaheadMenu/TypeaheadMenu.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Pagination = exports.Default = exports.CustomOption = exports.CustomChildren = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _TypeaheadMenu = _interopRequireDefault(require("./TypeaheadMenu"));

var _data = _interopRequireDefault(require("../../tests/data"));

var _utils = require("../../utils");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var _default = {
  title: 'Components/TypeaheadMenu',
  component: _TypeaheadMenu["default"]
};
exports["default"] = _default;
var defaultProps = {
  id: 'typeahead-menu',
  labelKey: 'name',
  options: _data["default"],
  text: ''
};

var Template = function Template(args) {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      minHeight: '300px'
    }
  }, /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      position: 'relative'
    }
  }, /*#__PURE__*/_react["default"].createElement(_TypeaheadMenu["default"], args)));
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = _objectSpread({}, defaultProps);
var CustomOption = Template.bind({});
exports.CustomOption = CustomOption;
CustomOption.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  options: [{
    customOption: true,
    name: 'custom option'
  }],
  text: 'custom option'
});
var Pagination = Template.bind({});
exports.Pagination = Pagination;
Pagination.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  options: [].concat((0, _toConsumableArray2["default"])(_data["default"].slice(0, 5)), [{
    paginationOption: true
  }])
});
var CustomChildren = Template.bind({});
exports.CustomChildren = CustomChildren;
CustomChildren.args = _objectSpread(_objectSpread({}, defaultProps), {}, {
  renderMenuItemChildren: function renderMenuItemChildren(option) {
    var name = (0, _utils.getOptionProperty)(option, 'name');
    var population = (0, _utils.getOptionProperty)(option, 'population');
    return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("div", null, name), /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement("small", null, "Population: ", population.toLocaleString())));
  }
});
