// react-bootstrap-typeahead/cjs/tests/helpers.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  noop: true,
  prepareSnapshot: true,
  generateSnapshots: true,
  HintProvider: true,
  getHint: true,
  findInput: true,
  getInput: true,
  findItems: true,
  getItems: true,
  findMenu: true,
  getMenu: true,
  findPaginator: true,
  getTokens: true,
  waitForOverlay: true,
  axe: true,
  userEvent: true
};
exports.HintProvider = void 0;
Object.defineProperty(exports, "axe", {
  enumerable: true,
  get: function get() {
    return _jestAxe.axe;
  }
});
exports.findInput = findInput;
exports.findItems = findItems;
exports.findMenu = findMenu;
exports.findPaginator = findPaginator;
exports.generateSnapshots = generateSnapshots;
exports.getHint = getHint;
exports.getInput = getInput;
exports.getItems = getItems;
exports.getMenu = getMenu;
exports.getTokens = getTokens;
exports.noop = noop;
exports.prepareSnapshot = prepareSnapshot;
Object.defineProperty(exports, "userEvent", {
  enumerable: true,
  get: function get() {
    return _userEvent["default"];
  }
});
exports.waitForOverlay = waitForOverlay;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _jestAxe = require("jest-axe");

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

var _testingReact = require("@storybook/testing-react");

Object.keys(_testingReact).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _testingReact[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _testingReact[key];
    }
  });
});

var _react2 = require("@testing-library/react");

Object.keys(_react2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _react2[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _react2[key];
    }
  });
});

var _Context = require("../core/Context");

var _userEvent = _interopRequireDefault(require("@testing-library/user-event"));

var _excluded = ["children"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

// eslint-disable-next-line @typescript-eslint/no-empty-function
function noop() {}

function prepareSnapshot(element) {
  return _reactTestRenderer["default"].create(element).toJSON();
}

function generateSnapshots(stories) {
  var composed = (0, _testingReact.composeStories)(stories);
  Object.entries(composed).forEach(function (_ref) {
    var _Story$parameters, _Story$parameters$sna;

    var _ref2 = (0, _slicedToArray2["default"])(_ref, 2),
        storyName = _ref2[0],
        Story = _ref2[1];

    if ((_Story$parameters = Story.parameters) !== null && _Story$parameters !== void 0 && (_Story$parameters$sna = _Story$parameters.snapshot) !== null && _Story$parameters$sna !== void 0 && _Story$parameters$sna.skip) return;
    test("".concat(storyName, " story renders snapshot"), /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              expect(prepareSnapshot( /*#__PURE__*/_react["default"].createElement(Story, null))).toMatchSnapshot();

            case 1:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    })));
  });
}

var HintProvider = function HintProvider(_ref4) {
  var children = _ref4.children,
      context = (0, _objectWithoutProperties2["default"])(_ref4, _excluded);
  return /*#__PURE__*/_react["default"].createElement(_Context.TypeaheadContext.Provider, {
    value: _objectSpread(_objectSpread({}, _Context.defaultContext), context)
  }, children);
};
/**
 * Finding elements
 */


exports.HintProvider = HintProvider;

function getHint(container) {
  return container.getElementsByClassName('rbt-input-hint')[0];
}

function findInput() {
  return _findInput.apply(this, arguments);
}

function _findInput() {
  _findInput = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
    var multiple,
        role,
        _args2 = arguments;
    return _regenerator["default"].wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            multiple = _args2.length > 0 && _args2[0] !== undefined ? _args2[0] : false;
            role = multiple ? 'textbox' : 'combobox';
            return _context2.abrupt("return", _react2.screen.findByRole(role));

          case 3:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  }));
  return _findInput.apply(this, arguments);
}

function getInput() {
  // Look for either the single- or multi-select case.
  return _react2.screen.queryByRole('combobox') || _react2.screen.queryByRole('textbox');
}

function findItems() {
  return _findItems.apply(this, arguments);
}

function _findItems() {
  _findItems = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3() {
    return _regenerator["default"].wrap(function _callee3$(_context3) {
      while (1) {
        switch (_context3.prev = _context3.next) {
          case 0:
            return _context3.abrupt("return", _react2.screen.findAllByRole('option'));

          case 1:
          case "end":
            return _context3.stop();
        }
      }
    }, _callee3);
  }));
  return _findItems.apply(this, arguments);
}

function getItems() {
  return _react2.screen.getAllByRole('option');
}

function findMenu() {
  return _findMenu.apply(this, arguments);
}

function _findMenu() {
  _findMenu = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4() {
    return _regenerator["default"].wrap(function _callee4$(_context4) {
      while (1) {
        switch (_context4.prev = _context4.next) {
          case 0:
            return _context4.abrupt("return", _react2.screen.findByRole('listbox'));

          case 1:
          case "end":
            return _context4.stop();
        }
      }
    }, _callee4);
  }));
  return _findMenu.apply(this, arguments);
}

function getMenu() {
  return _react2.screen.queryByRole('listbox');
}

function findPaginator() {
  return _findPaginator.apply(this, arguments);
}

function _findPaginator() {
  _findPaginator = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5() {
    var items;
    return _regenerator["default"].wrap(function _callee5$(_context5) {
      while (1) {
        switch (_context5.prev = _context5.next) {
          case 0:
            _context5.next = 2;
            return findItems();

          case 2:
            items = _context5.sent;
            return _context5.abrupt("return", items[items.length - 1]);

          case 4:
          case "end":
            return _context5.stop();
        }
      }
    }, _callee5);
  }));
  return _findPaginator.apply(this, arguments);
}

function getTokens(container) {
  return container.getElementsByClassName('rbt-token');
}

function waitForOverlay() {
  return _waitForOverlay.apply(this, arguments);
}

function _waitForOverlay() {
  _waitForOverlay = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6() {
    return _regenerator["default"].wrap(function _callee6$(_context6) {
      while (1) {
        switch (_context6.prev = _context6.next) {
          case 0:
            _context6.next = 2;
            return findMenu();

          case 2:
          case "end":
            return _context6.stop();
        }
      }
    }, _callee6);
  }));
  return _waitForOverlay.apply(this, arguments);
}
