// react-bootstrap-typeahead/cjs/components/Token/Token.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Static = exports.Interactive = exports.Disabled = exports.Anchor = void 0;

var _react = _interopRequireDefault(require("react"));

var _Token = _interopRequireDefault(require("./Token"));

var _utils = require("../../utils");

/* eslint-disable sort-keys,import/no-extraneous-dependencies */
var _default = {
  title: 'Components/Token',
  component: _Token["default"]
};
exports["default"] = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react["default"].createElement(_Token["default"], args);
};

var Interactive = Template.bind({});
exports.Interactive = Interactive;
Interactive.args = {
  children: 'This is an interactive token',
  onRemove: _utils.noop
};
var Static = Template.bind({});
exports.Static = Static;
Static.args = {
  children: 'This is a static token',
  readOnly: true
};
var Anchor = Template.bind({});
exports.Anchor = Anchor;
Anchor.args = {
  children: 'This is a link token',
  href: '#',
  readOnly: true
};
var Disabled = Template.bind({});
exports.Disabled = Disabled;
Disabled.args = {
  children: 'This is a disabled token',
  disabled: true
};
