// react-bootstrap-typeahead/es/components/Overlay/Overlay.stories.js
import _extends from "@babel/runtime/helpers/extends";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";

/* eslint-disable sort-keys,import/no-extraneous-dependencies */
import React, { useState } from 'react';
import Overlay from './Overlay';
import Menu from '../Menu';
import { BaseMenuItem } from '../MenuItem';
export default {
  title: 'Components/Overlay',
  component: Overlay
};

var Template = function Template(args) {
  var _useState = useState(null),
      _useState2 = _slicedToArray(_useState, 2),
      referenceElement = _useState2[0],
      setReferenceElement = _useState2[1];

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
    ref: setReferenceElement,
    style: {
      border: '1px solid red',
      padding: '4px'
    }
  }, "Reference element"), /*#__PURE__*/React.createElement(Overlay, _extends({}, args, {
    referenceElement: referenceElement
  }), function (menuProps) {
    return /*#__PURE__*/React.createElement(Menu, _extends({}, menuProps, {
      id: "overlay-story"
    }), /*#__PURE__*/React.createElement(BaseMenuItem, {
      role: "option"
    }, "This is the menu"));
  }));
};

export var Default = Template.bind({});
Default.args = {
  isMenuShown: true,
  positionFixed: true
};
