// react-bootstrap-typeahead/types/components/Menu/Menu.stories.js
import React from 'react';
import Menu from './Menu';
import MenuItem from '../MenuItem';
import { noop } from '../../utils';
const options = [{ label: 'Item 1' }, { label: 'Item 2' }, { label: 'Item 3' }];
export default {
    title: 'Components/Menu',
    component: Menu,
};
const defaultProps = {
    inputHeight: 0,
    scheduleUpdate: noop,
    text: '',
};
const children = options.map((o, idx) => (React.createElement(MenuItem, { key: o.label, option: o, position: idx }, o.label)));
const Template = (args) => (React.createElement(Menu, { ...args, style: {
        ...args.style,
        position: 'relative',
    } }));
export const Default = Template.bind({});
Default.args = {
    ...defaultProps,
    children,
    id: 'default-menu',
};
export const Empty = Template.bind({});
Empty.args = {
    ...defaultProps,
    id: 'empty-menu',
};
export const HeaderAndDivider = Template.bind({});
HeaderAndDivider.args = {
    ...defaultProps,
    children: (React.createElement(React.Fragment, null,
        React.createElement(Menu.Header, null, "This is a menu header"),
        React.createElement(Menu.Divider, null),
        React.createElement(MenuItem, { option: options[0], position: 0 }, options[0].label))),
    id: 'header-and-divider',
};
